src/Service/DTOCacheTagger.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use DateInterval;
+use ReflectionClass as ReflectionClassAlias;
+use Symfony\Contracts\Cache\ItemInterface;
+
+class DTOCacheTagger
+{
+    public function __construct(protected EntityMetadata $entityMetadata)
+    {
+    }
+
+    /**
+     * Tag a cache item with all the relationships and data for a DTO
+     */
+    public function tag(ItemInterface $item, object $dto): void
+    {
+        $tags = $this->getTagsForDto($dto);
+        $item->tag($tags);
+        $item->expiresAfter(DateInterval::createFromDateString('1 day'));
```

Comment:
🤷 It's more of a safety valve in case the cache gets into really bad shape or if something isn't getting invalidated. I'm hopping to remove it at some point and just let the tag expiration do this work.